eth/handler.go

```diff
@@ -399,3 +410,28 @@ func (h *handler) txBroadcastLoop() {
 		}
 	}
 }
+
+// missingBodyLoop listens to the MissingBody event in Slice and calls the blockAnnounces.
+func (h *handler) missingBodyLoop() {
+	defer h.wg.Done()
+	for {
+		select {
+		case header := <-h.missingBodyCh:
+			// Check if any of the peers have the body
+			for _, peer := range h.peers.peers {
+				if peer.KnownBlock(header.Hash()) {
+					log.Debug("Fetching the missing body from", "peer", peer.ID(), "hash", header.Hash(), "number", header.NumberU64())
+					h.blockFetcher.Notify(peer.ID(), header.Hash(), header.NumberU64(), time.Now(), peer.RequestOneHeader, peer.RequestBodies)
+					continue
+				}
+			}
+			// If none of the peers have the block, we can ask the peer with highest number for the body
+			highestPeer := h.peers.peerWithHighestNumber()
```

Comment:
Isn't this case already handled in the above peer loop?

Reply:
removed it and added a warn